Polyhedron/demo/Polyhedron/Scene_item.cpp

```diff
@@ -116,276 +116,33 @@ void Scene_item::select(double /*orig_x*/,
-            return program;
-        }
-        break;
-    case PROGRAM_WITH_TEXTURED_EDGES:
-        if( shader_programs[PROGRAM_WITH_TEXTURED_EDGES])
-        {
-            return shader_programs[PROGRAM_WITH_TEXTURED_EDGES];
-        }
-        else
-        {
-            QOpenGLShaderProgram *program = new QOpenGLShaderProgram(viewer);
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Vertex,":/cgal/Polyhedron_3/resources/shader_with_textured_edges.v" ))
-            {
-                std::cerr<<"adding vertex shader FAILED"<<std::endl;
-            }
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Fragment,":/cgal/Polyhedron_3/resources/shader_with_textured_edges.f" ))
-            {
-                std::cerr<<"adding fragment shader FAILED"<<std::endl;
-            }
-            program->link();
-            shader_programs[PROGRAM_WITH_TEXTURED_EDGES] = program;
-            return program;
-
-        }
-        break;
-    case PROGRAM_INSTANCED:
-        if( shader_programs[PROGRAM_INSTANCED])
-        {
-            return shader_programs[PROGRAM_INSTANCED];
-        }
-        else
-        {
-            QOpenGLShaderProgram *program = new QOpenGLShaderProgram(viewer);
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Vertex,":/cgal/Polyhedron_3/resources/shader_instanced.v" ))
-            {
-                std::cerr<<"adding vertex shader FAILED"<<std::endl;
-            }
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Fragment,":/cgal/Polyhedron_3/resources/shader_with_light.f" ))
-            {
-                std::cerr<<"adding fragment shader FAILED"<<std::endl;
-            }
-            program->link();
-            shader_programs[PROGRAM_INSTANCED] = program;
-            return program;
-
-        }
-        break;
-    case PROGRAM_INSTANCED_WIRE:
-        if( shader_programs[PROGRAM_INSTANCED_WIRE])
-        {
-            return shader_programs[PROGRAM_INSTANCED_WIRE];
-        }
-        else
-        {
-            QOpenGLShaderProgram *program = new QOpenGLShaderProgram(viewer);
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Vertex,":/cgal/Polyhedron_3/resources/shader_instanced.v" ))
-            {
-                std::cerr<<"adding vertex shader FAILED"<<std::endl;
-            }
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Fragment,":/cgal/Polyhedron_3/resources/shader_without_light.f" ))
-            {
-                std::cerr<<"adding fragment shader FAILED"<<std::endl;
-            }
-            program->link();
-            shader_programs[PROGRAM_INSTANCED_WIRE] = program;
-            return program;
-
-        }
-        break;
-    default:
-        std::cerr<<"ERROR : Program not found."<<std::endl;
-        return 0;
-    }
+    if(viewer == 0)
+        viewer = dynamic_cast<CGAL::Three::Viewer_interface*>(*QGLViewer::QGLViewerPool().begin());
+    return viewer->getShaderProgram(name);
```

Comment:
`Scene_item::getShaderProgram` is kept for compatibility with the existing item classes. It just call `viewer->getShaderProgram(name)`.

It would be best to change all calls to `getShaderProgram(..)` in items to `viewer->getShaderProgram(..)` but I wanted to minimize the patch size.